packages/connect/src/types/api/firmwareUpdate.ts

```diff
@@ -15,17 +16,29 @@ export const FirmwareUpdate = Type.Union([
     }),
 ]);
 
-export type FirmwareUpdateResponse =
-    | {
-          check:
-              | 'mismatch' //  firmware is not legitimate
-              | 'valid' // ok
-              | 'omitted'; // custom fw binary, or maybe older fw
-      }
-    | {
-          check: 'other-error';
-          checkError: string; // unable to carry out the check due to a non-related error such as disconnected device
-      };
+type FirmwareUpdateDetails = {
+    versionCheck: boolean;
+    bootloaderVersion: VersionArray;
+    installedVersion: VersionArray;
+    latestVersion: VersionArray;
+    binaryVersion: VersionArray;
+    releaseVersion?: VersionArray;
+    isLatest: boolean;
+};
```

Comment:
I wouldn't mind adding some comments here explaining what these fields mean. Actually, only one aspect is not obvious at first glance - is the <something>-version that has just been installed, or that had been in the device before installation or that was available for installation?

Reply:
good point, https://github.com/trezor/trezor-suite/pull/16421/commits/710fc27d1f9c7d872a2d4d032be9fa78dec226c1